Wikipedia/Code/TalkPageFetcher.swift

```diff
@@ -64,5 +64,80 @@ class TalkPageFetcher: Fetcher {
             }
         }
     }
-
+    
+    func subscribeToTopic(talkPageTitle: String, siteURL: URL, topic: String, shouldSubscribe: Bool, completion: @escaping (Result<Bool, Error>) -> Void) {
+        
+        guard let title = talkPageTitle.denormalizedPageTitle else {
+            completion(.failure(RequestError.invalidParameters))
+            return
+        }
+                
+        var params = ["action": "discussiontoolssubscribe",
+                      "page": title,
+                      "format": "json",
+                      "commentname": topic,
+                      "formatversion": "2"
+        ]
+        
+        if shouldSubscribe {
+            params["subscribe"] = "1"
+        }
+        
+        performTokenizedMediaWikiAPIPOST(to: siteURL, with: params, reattemptLoginOn401Response: true) { result, httpResponse, error in
+            if let error = error {
+                completion(.failure(error))
+                return
+            }
+            
+            if let resultError = result?["error"] as? [String: Any],
+               let info = resultError["info"] as? String {
+                completion(.failure(RequestError.api(info)))
+                return
+            }
+            
+            if let resultSuccess = result?["discussiontoolssubscribe"] as? [String: Any],
+                let didSubscribe = resultSuccess["subscribe"] as? Bool {
+                completion(.success(didSubscribe))
+                return
+            }
+            completion(.failure(RequestError.unexpectedResponse))
+        }
+    }
+    
+    func getAllSubscriptions(siteURL: URL, topics: [String], completion: @escaping (Result<[String], Error>) -> Void) {
```

Comment:
Actually now that we know we'll only be sending back the subscribed topics, maybe we should rename this to something like:

```
func getSubscribedTopics(from topics: [String], siteURL: URL, completion: @escaping (Result<[String], Error>) -> Void) {
```